x/incentives/keeper/store.go

```diff
@@ -118,6 +118,25 @@ func (k Keeper) GetAllGroups(ctx sdk.Context) ([]types.Group, error) {
 	return osmoutils.GatherValuesFromStorePrefix(ctx.KVStore(k.storeKey), types.KeyPrefixGroup, k.ParseGroupFromBz)
 }
 
+// GetAllGroupsGauges first gets all groups, iterates through them, and pulls the gauge from each.
+func (k Keeper) GetAllGroupsGauges(ctx sdk.Context) ([]types.Gauge, error) {
+	groups, err := k.GetAllGroups(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var gauges []types.Gauge
+	for _, group := range groups {
+		gauge, err := k.GetGaugeByID(ctx, group.GroupGaugeId)
+		if err != nil {
+			return nil, err
+		}
+		gauges = append(gauges, *gauge)
+	}
```

Comment:
I think it would eventually make sense to move towards iterating over group store index with regular store iterator

As we iterate and call `groupIter.Next()` and `groupIter.GetValue()`, we can parse the group, retrieve the `GroupGaugeID` and then read the associated gauge.

This will have the following benefits:
1. Reduce the need for doing loop iterations
2. Return a wrapper data structure where `Group` is packaged together with group `Gauge`.

This would be useful for queries because, as a client, it makes perfect sense to retrieve groups together with their gauges

I'm not blocking on this comment but I think we might want to eventually move to this for queries

Reply:
Used prefix store iterator here https://github.com/osmosis-labs/osmosis/pull/6565/commits/b1d79ca2124e0d510964d76a408de7be682602a4

Reply:
Did not return Group with Gauge, but this can be easily added in a follow on PR